homeassistant/components/tado/coordinator.py

```diff
@@ -0,0 +1,433 @@
+            duration,
+            device_type,
+            mode,
+            fan_speed,
+            swing,
+            fan_level,
+            vertical_swing,
+            horizontal_swing,
+        )
+
+        try:
+            await self.hass.async_add_executor_job(
+                self._tado.set_zone_overlay,
+                zone_id,
+                overlay_mode,
+                temperature,
+                duration,
+                device_type,
+                "ON",
+                mode,
+                fan_speed,
+                swing,
+                fan_level,
+                vertical_swing,
+                horizontal_swing,
+            )
+
+        except RequestException as err:
+            raise UpdateFailed(f"Error setting Tado overlay: {err}") from err
+
+        await self._update_zone(zone_id)
+
+    async def set_zone_off(self, zone_id, overlay_mode, device_type="HEATING"):
+        """Set a zone to off."""
+        try:
+            await self.hass.async_add_executor_job(
+                self._tado.set_zone_overlay,
+                zone_id,
+                overlay_mode,
+                None,
+                None,
+                device_type,
+                "OFF",
+            )
+        except RequestException as err:
+            raise UpdateFailed(f"Error setting Tado overlay: {err}") from err
+
+        await self._update_zone(zone_id)
+
+    async def set_temperature_offset(self, device_id, offset):
+        """Set temperature offset of device."""
+        try:
+            await self.hass.async_add_executor_job(
+                self._tado.set_temp_offset, device_id, offset
+            )
+        except RequestException as err:
+            raise UpdateFailed(f"Error setting Tado temperature offset: {err}") from err
+
+    async def set_meter_reading(self, reading: int) -> dict[str, Any]:
+        """Send meter reading to Tado."""
+        dt: str = datetime.now().strftime("%Y-%m-%d")
+        if self._tado is None:
+            raise HomeAssistantError("Tado client is not initialized")
+
+        try:
+            return await self.hass.async_add_executor_job(
+                self._tado.set_eiq_meter_readings, dt, reading
+            )
+        except RequestException as err:
+            raise UpdateFailed(f"Error setting Tado meter reading: {err}") from err
+
+
+class TadoMobileDeviceUpdateCoordinator(DataUpdateCoordinator[dict[str, dict]]):
+    """Class to manage the mobile devices from Tado via PyTado."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        tado: Tado,
+    ) -> None:
+        """Initialize the Tado data update coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=SCAN_MOBILE_DEVICE_INTERVAL,
+        )
+        self._tado = tado
+        self.data: dict[str, dict] = {
+            "mobile_device": {},
+        }
```

Comment:
Why do we still wrap the data in a dict? We can just put the data directly back so we don't have to `["mobile_device"]` it